pkg/hooks/hooks.go

```diff
@@ -94,29 +96,85 @@ func runDirectory(dirpath string, environment []string, logger logging.Logger) e
 		if f.IsDir() {
 			continue
 		}
-		logger.WithField("path", fullpath).Infof("Executing hook %s", f.Name())
-		cmd := exec.Command(fullpath)
-		hookOut := &bytes.Buffer{}
-		cmd.Stdout = hookOut
-		cmd.Stderr = hookOut
-		cmd.Env = environment
-		err := cmd.Run()
-		if err != nil {
-			logger.WithErrorAndFields(err, logrus.Fields{
-				"path":   fullpath,
-				"output": hookOut.String(),
-			}).Warnf("Could not execute hook %s", f.Name())
-		} else {
-			logger.WithFields(logrus.Fields{
-				"path":   fullpath,
-				"output": hookOut.String(),
-			}).Debugln("Executed hook")
+
+		timeout := 5 * time.Second
+		h := Hook{fullpath, f.Name(), &timeout, environment, logger}
+
+		err := h.RunWithTimeout()
```

Comment:
Although there's no way to successfully return an error from `RunWithTimeout` that's _not_ a `HookTimeoutError`, it would be nice to see a `else if err != nil` case below